static/app/views/performance/newTraceDetails/traceDrawer/details/transaction/index.tsx

```diff
@@ -134,6 +149,7 @@ export function TransactionNodeDetails({
         />
         <AdditionalData event={event} />
         <Measurements event={event} location={location} organization={organization} />
+        {cacheMetrics[0] && <CacheMetrics cacheMetrics={cacheMetrics[0]} />}
```

Comment:
Why only cacheMetrics[0]? Do we always expect it to be a single metric or should we instead map and display all of these metric here?

Reply:
Yeah we always expect a single metric. The Discover endpoint just always returns an array of results by design

Reply:
For clarification, `cacheMetrics` is of type ` Pick<SpanMetricsResponse, 'avg(cache.item_size)' | 'cache_miss_rate()'>[]`.
We only expect one element in this array (as we're querying for a single transaction), containing the `avg(cache.item_size)` and `cache_miss_rate()` metrics for that transaction.

The only case where we have multiple results is if there are multiple transactions with the same name, but that shouldn't be possible.

Reply:
That's fair, this pattern is used extensively in modules, so it was more of an oversight. I'm okay with rendering all of them too, i like how it has that visual explanation if something weird is happening as you mentioned.
Will update the PR!

Reply:
Oh yeah, we do this a ton all over sentry and it's just something I want us to be more mindful about.